core/src/main/java/org/infinispan/factories/InternalCacheFactory.java

```diff
@@ -52,6 +73,39 @@
       }
    }
 
+   private boolean allowSimpleCache(Configuration configuration) {
+      if (!configuration.allowSimple()) {
+         return false;
+      }
+      ClusteringConfiguration clustering = configuration.clustering();
+      TransactionConfiguration transaction = configuration.transaction();
+      CustomInterceptorsConfiguration interceptors = configuration.customInterceptors();
+      PersistenceConfiguration persistence = configuration.persistence();
+      InvocationBatchingConfiguration invocationBatching = configuration.invocationBatching();
+      IndexingConfiguration indexing = configuration.indexing();
+      StoreAsBinaryConfiguration storeAsBinary = configuration.storeAsBinary();
+      JMXStatisticsConfiguration jmxStatistics = configuration.jmxStatistics();
+      // TODO: eventually support transactional caches as well
+      if (clustering != null && clustering.cacheMode().isClustered()) {
```

Comment:
I find the if-else syntax more compact, but if you prefer it that way, I won't object.